thrust/thrust/system/cuda/detail/transform.h

```diff
@@ -202,6 +205,17 @@ OutputIt _CCCL_API _CCCL_FORCEINLINE cub_transform_many(
     policy, get<0>(firsts).get_iterator_tuple(), result, num_items, transform_op.underlying_function());
 }
 
+template <class Derived, class Offset, class... InputIts, class OutputIt, class TransformOp>
+OutputIt _CCCL_API _CCCL_FORCEINLINE cub_transform_many(
+  execution_policy<Derived>& policy,
+  ::cuda::std::tuple<::cuda::zip_iterator<InputIts...>> firsts,
+  OutputIt result,
+  Offset num_items,
+  ::cuda::zip_function<TransformOp> transform_op)
```

Comment:
I believe we should reuse the `__is_zip_iterator` and `__is_zip_function `traits that are defined in `__fwd/zip_iterator.h`

That would allow us to also catch accidental issues like mixing thrust and cuda facilities